qt/python/mantidqt/utils/qt/testing/run_test_app.py

```diff
@@ -0,0 +1,160 @@
+#  This file is part of the mantid workbench.
+#
+#  Copyright (C) 2017 mantidproject
+#
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""A small command line utility that can be used to manually test a widget in isolation.
+
+Usage:
+
+    python run_test_app.py qualified.name.of.SomeWidget [--script qualified.name.of.some_script]
+
+The widget (and optional script) should be in the location form where python could import it as in:
+
+    form qualified.name.of import SomeWidget
+
+The widget is created by calling SomeWidget() constructor without arguments. If __init__ has arguments or
+the widget needs to be specially prepared any function returning an instance of SomeWidget can be used instead.
+
+Optional --script argument expects a name of a function which takes a widget as its single argument.
+If specified the script is imported and run after the widget is created.
+
+"""
+from __future__ import absolute_import, print_function
+import sys
+import traceback
+from qtpy.QtCore import Qt, QTimer, QPoint
+from qtpy.QtGui import QContextMenuEvent
+from qtpy.QtWidgets import QApplication, QAction
+from mantidqt.utils.qt.plugins import setup_library_paths
+
+
+class Tester(object):
+
+    def __init__(self, widget):
+        self.widget = widget
+        self.input_text = None
+        self.timer = None
+
+    def __call__(self):
+        widget = self.widget
+        items = widget.tree.findItems('Squares v.1', Qt.MatchExactly | Qt.MatchRecursive)
```

Comment:
This module seems to be the general tester for running a standalone widget. What's the reason for the calls to functionality related to the algorithm selector widget?